buildlib/pr/mad_tests.yml

```diff
@@ -0,0 +1,112 @@
+jobs:
+  - job: SetupServer
+    displayName: Setup Server
+    pool:
+      name: MLNX
+      demands: mad_server
+    workspace:
+      clean: outputs
+    steps:
+      - checkout: self
+        clean: true
+        fetchDepth: 100
+        retryCountOnTaskFailure: 5
+      - task: Bash@3
+        name: Set_Vars
+        inputs:
+          targetType: "inline"
+          script: |
+            source ./buildlib/tools/test_mad.sh
+            set_vars
+        displayName: Set Vars
+      - bash: |
+          source ./buildlib/tools/test_mad.sh
+          build_ucx_in_docker
+          docker_run_srv
+        displayName: Setup Server
+
+  - job: SetupClient
+    displayName: Setup Client
+    pool:
+      name: MLNX
+      demands: mad_client
+    workspace:
+      clean: outputs
+    steps:
+      - checkout: self
+        clean: true
+        fetchDepth: 100
+        retryCountOnTaskFailure: 5
+      - bash: |
+          source ./buildlib/tools/test_mad.sh
+          build_ucx
+        displayName: Setup Client
+
+  - job: TestLid
+    dependsOn:
+      - SetupServer
+      - SetupClient
+    displayName: Test Lid
+    timeoutInMinutes: 10
+    pool:
+      name: MLNX
+      demands: mad_client
+    variables:
+      LID: $[ dependencies.SetupServer.outputs['Set_Vars.LID'] ]
+      HCA: $[ dependencies.SetupServer.outputs['Set_Vars.HCA'] ]
+    steps:
+      - checkout: none
+      - bash: |
+          source ./buildlib/tools/test_mad.sh
+          run_mad_test lid $(LID)
```

Comment:
we might as well either pass `lid:$(LID)` or `guid:$(GUID)` to simplify argument handling in the called function?